api/ingress/v1alpha1/ngrok_common.go

```diff
@@ -428,3 +430,81 @@ func (amazon *EndpointOAuthAmazon) ToNgrok(clientSecret *string) *ngrok.Endpoint
 	}
 	return mod
 }
+
+type EndpointPolicies struct {
+	// Determines if the policy will be applied to traffic
+	Enabled *bool `json:"enabled,omitempty"`
+	// Policies for inbound traffic
+	Inbound []EndpointPolicy `json:"inbound,omitempty"`
+	// Policies for outbound traffic
+	Outbound []EndpointPolicy `json:"outbound,omitempty"`
+}
+
+type EndpointPolicy struct {
+	// Expressions
+	Expressions []string `json:"expressions,omitempty"`
+	// Actions
+	Actions []EndpointAction `json:"actions,omitempty"`
+	// Name
+	Name string `json:"name,omitempty"`
+}
+
+type EndpointAction struct {
+	Type string `json:"type,omitempty"`
+	// +kubebuilder:validation:Schemaless
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +kubebuilder:validation:Type=object
+	Config json.RawMessage `json:"config,omitempty"`
+}
+
+func (policies *EndpointPolicies) ToNgrok() *ngrok.EndpointPolicies {
+	if policies == nil {
+		return nil
+	}
+
+	var inbound []ngrok.EndpointPolicy
+	for _, policy := range policies.Inbound {
+		p := policy
+		inbound = append(inbound, *p.ToNgrok())
```

Comment:
should you check for `nil` here?

Reply:
and further down?

Reply:
Thanks for the catch! Fixed here: https://github.com/ngrok/kubernetes-ingress-controller/pull/334/commits/21c78946a6141ceb2b4f5f6d2603a89b015a5662